### PR TITLE
fix: Bid + Registration flows modal header

### PR DESCRIPTION
**Pull request by ds300**
This PR resolves [CX-1181]

### Description

This updates the bid flow and the registration flow to use our standard modal header style.

#### Bid Flow
<details>
  <summary>SelectMaxBid</summary>

</details>
<details>
  <summary>ConfirmBid</summary>

</details>
<details>
  <summary>BidResult</summary>
  

</details>

#### Registration flow
<details>
  <summary>Registration</summary>

</details>
<details>
  <summary>RegistrationResult</summary>

</details>

#### Shared
<details>
  <summary>CreditCardForm</summary>

</details>
<details>
  <summary>BillingAddress</summary>

</details>
<details>
  <summary>CountrySelect</summary>
  

</details>

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-1181]: https://artsyproduct.atlassian.net/browse/CX-1181

### CODE REVIEW

File: src/lib/Components/Bidding/Screens/SelectMaxBid.tsx [hunk 1]
```diff
@@ -121,3 +117,41 @@ export const MaxBidScreen = createRefetchContainer(
...
+          artworkID,
+          saleID,
+        }}
+        render={renderWithLoadProgress<SelectMaxBidQuery["response"]>((props) => (
+          <SelectMaxBidContainer me={props.me!} sale_artwork={props.artwork!.sale_artwork!} navigator={navigator} />
+        ))}
+      />
+    </Flex>
+  )
+}
```
Comment (ds300) [inline src/lib/Components/Bidding/Screens/SelectMaxBid.tsx:157]: the query renderer for this flow used to live _above_ the navigator, which created an extra layer of indirection for no good reason, so I simplified this to make it more like our other screens.

File: src/lib/Components/Bidding/Screens/__tests__/BillingAddress-tests.tsx [hunk 2]
```diff
@@ -83,12 +83,12 @@ it("updates the validation for country when coming back from the select country
...
   component.root.findByType(Button).props.onPress()
 
-  expect(component.root.findAllByType(Sans)[0].props.children).toEqual("This field is required")
+  expect(component.root.findAllByType(Sans)[1].props.children).toEqual("This field is required")
 
   selectCountry(component, fakeNavigator, billingAddress.country)
 
   // The <Sans12> instances in the BillingAddress screen display error messages
-  expect(component.root.findAllByType(Sans).length).toEqual(1)
+  expect(component.root.findAllByType(Sans).length).toEqual(2)
```
Comment (ds300) [inline src/lib/Components/Bidding/Screens/__tests__/BillingAddress-tests.tsx:91]: such brittle tests 🙈

**Comment by samchieng** [comment]
@ds300 these look great!! I don't think the bid result needs a title, and the only change I would suggest is a slight copy tweak for 2 of them:

* Your credit card -> Add credit card
* Your billing address -> Add billing address

This is just to keep it consistent with the other titles that start with an action verb as instructions.

File: src/lib/AppRegistry.tsx [hunk 3]
```diff
@@ -325,16 +325,12 @@ export const modules = defineModules({
...
-      : reactModule(BidFlowQueryRenderer, {
-          alwaysPresentModally: true,
-          hasOwnModalCloseButton: true,
-        }),
+  AuctionBidArtwork: reactModule(BidFlow, {
+    alwaysPresentModally: true,
+    hasOwnModalCloseButton: true,
+    fullBleed: true,
+  }),
+  // BidFlow is used by ARBidFlowViewController. Once that has been deleted we can delete this too.
```
[the remark this answers is not in the document]
Comment (ds300) [reply src/lib/AppRegistry.tsx]: added the registration flow screens to this PR and deleted all this old stuff

File: Artsy/App/ARScreenPresenterModule.h [hunk 4]
```diff
@@ -8,7 +8,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ARScreenPresenterModule : NSObject <RCTBridgeModule>
 
-+ (UIViewController *)loadAuctionRegistrationWithID:(NSString *)auctionID skipBidFlow:(BOOL)skipBidFlow;
++ (UIViewController *)loadWebViewAuctionRegistrationWithID:(NSString *)auctionID;
```
Comment (ds300) [inline Artsy/App/ARScreenPresenterModule.h:11]: this was only being used by the live auction interface with `skipBidFlow` set to `YES`. Despite the name, `skipBidFlow` seemed to simply force the auction registration flow to load in a web view. So I removed the option and turned this into a method that always returns the web version.

File: src/lib/Components/Bidding/Screens/SelectMaxBid.tsx [hunk 5]
```diff
@@ -121,3 +124,38 @@ export const MaxBidScreen = createRefetchContainer(
     }
   `
 )
+
+export const SelectMaxBidQueryRenderer: React.FC<{
+  artworkID: string
+  saleID: string
+  navigator: NavigatorIOS
+}> = ({ artworkID, saleID, navigator }) => {
+  // TODO: artworkID can be nil, so omit that part of the query if it is.
```
Comment (brainbicycle) [inline src/lib/Components/Bidding/Screens/SelectMaxBid.tsx:126]: **nitpick (non-blocking):** is this comment still necessary now that artworkID is non-optional?

File: src/lib/Components/Bidding/Screens/ConfirmBid/index.tsx [hunk 6]
```diff
@@ -522,10 +510,10 @@ export class ConfirmBid extends React.Component<ConfirmBidProps, ConfirmBidState
               <BidInfoRow
                 label="Max bid"
                 value={this.selectedBid().display}
-                onPress={isLoading ? () => null : () => this.goBackToSelectMaxBid()}
+                onPress={isLoading ? () => null : () => this.props.navigator?.pop()}
               />
 
-              {requiresPaymentInformation ? (
+              {requiresPaymentInformation || true ? (
```
Comment (brainbicycle) [inline src/lib/Components/Bidding/Screens/ConfirmBid/index.tsx]: **question:** isn't this always true? why have the check?

File: src/lib/AppRegistry.tsx [hunk 7]
```diff
@@ -325,16 +325,11 @@ export const modules = defineModules({
...
-        })
-      : reactModule(BidFlow, {
-          alwaysPresentModally: true,
-          hasOwnModalCloseButton: true,
-        }),
+  AuctionBidArtwork: reactModule(BidFlow, {
+    alwaysPresentModally: true,
+    hasOwnModalCloseButton: true,
+    fullBleed: true,
+  }),
```
Comment (brainbicycle) [inline src/lib/AppRegistry.tsx:307]: 🎉

File: src/lib/AppRegistry.tsx [hunk 8]
```diff
@@ -315,16 +315,11 @@ export const modules = defineModules({
...
-    Platform.OS === "ios"
-      ? nativeModule({
-          alwaysPresentModally: true,
-          fullBleed: true,
-        })
-      : reactModule(RegistrationFlowQueryRenderer, {
-          alwaysPresentModally: true,
-          hasOwnModalCloseButton: true,
-        }),
+  AuctionRegistration: reactModule(RegistrationFlow, {
```
Comment (brainbicycle) [inline src/lib/AppRegistry.tsx:298]: ✨

File: src/lib/Components/Bidding/Screens/ConfirmBid/index.tsx (hunk 6 above)
Comment (brainbicycle) [reply src/lib/Components/Bidding/Screens/ConfirmBid/index.tsx]: yes I am missing that is was test logic removed later on 🤦

**Review by brainbicycle — APPROVED** [review]
LGTM! Merge when ready

**Pull request merged**